test/test_file.cpp

```diff
@@ -431,81 +496,33 @@ TEST(File_Move)
     CHECK_NOT(file_2.is_attached());
 }
 
-#if 0
-TEST(File_PreallocResizing)
-{
-    TEST_PATH(path);
-    File file(path, File::mode_Write);
-    CHECK(file.is_attached());
-    // we cannot test this with encryption...prealloc always allocates a full page
-    file.prealloc(0); // this is allowed
-    CHECK_EQUAL(file.get_size(), 0);
-    file.prealloc(100);
-    CHECK_EQUAL(file.get_size(), 100);
-    file.prealloc(50);
-    CHECK_EQUAL(file.get_size(), 100); // prealloc does not reduce size
-
-    // To expose the preallocation bug, we need to iterate over a large numbers, less than 4096.
-    // If the bug is present, we will allocate additional space to the file on every call, but if it is
-    // not present, the OS will preallocate 4096 only on the first call.
-    constexpr size_t init_size = 2048;
-    constexpr size_t dest_size = 3000;
-    for (size_t prealloc_space = init_size; prealloc_space <= dest_size; ++prealloc_space) {
-        file.prealloc(prealloc_space);
-        CHECK_EQUAL(file.get_size(), prealloc_space);
-    }
-
-#if REALM_PLATFORM_APPLE
-    int fd = ::open(path.c_str(), O_RDONLY);
-    CHECK(fd >= 0);
-    struct stat statbuf;
-    CHECK(fstat(fd, &statbuf) == 0);
-    size_t allocated_size = statbuf.st_blocks;
-    CHECK_EQUAL(statbuf.st_size, dest_size);
-    CHECK(!int_multiply_with_overflow_detect(allocated_size, S_BLKSIZE));
-
-    // When performing prealloc, the OS has the option to preallocate more than the requeted space
-    // but we need to check that the preallocated space is within a reasonable bound.
-    // If space is being incorrectly preallocated (growing on each call) then we will have more than 3000KB
-    // of preallocated space, but if it is being allocated correctly (only when we need to expand) then we'll have
-    // a multiple of the optimal file system I/O operation (`stat -f %k .`) which is 4096 on HSF+.
-    // To give flexibility for file system prealloc implementations we check that the preallocated space is within
-    // at least 16 times the nominal requested size.
-    CHECK_LESS(allocated_size, 4096 * 16);
-
-    CHECK(::close(fd) == 0);
-#endif
-}
-#endif
-
 TEST(File_PreallocResizingAPFSBug)
 {
     TEST_PATH(path);
     File file(path, File::mode_Write);
     CHECK(file.is_attached());
-    file.write("aaaaaaaaaaaaaaaaaaaa"); // 20 a's
+    file.write(0, "aaaaaaaaaaaaaaaaaaaa"); // 20 a's
     // calling prealloc on a newly created file would sometimes fail on APFS with EINVAL via fcntl(F_PREALLOCATE)
     // this may not be the only way to trigger the error, but it does seem to be timing dependant.
     file.prealloc(100);
     CHECK_EQUAL(file.get_size(), 100);
 
     // let's write past the first prealloc block (@ 4096) and verify it reads correctly too.
-    file.write("aaaaa");
+    // FIXME: what is this write trying to do?
+    file.write(20, "aaaaa");
```

Comment:
I think this was relying on the previous call to prealloc to set the file position to the end of the file. If so, then this should be updated to `file.write(file.size(), "aaaaa")`